### PR TITLE
tests: Fix scratch buffer size mangement

### PR DESCRIPTION
**Pull request by arno-lunarg**
Closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/7496

### CODE REVIEW

**Comment by spencer-lunarg** [comment]
> closes #7496

This was also happening on tests like `NegativeRayTracing.ScratchBufferBadMemory` (see bottom of issue)

(will confirm tomorrow this fixes the issue for this test)

**Comment by arno-lunarg** [comment]
Ah I did not read the issue til the end, will update those tests

**Review by spencer-lunarg — APPROVED** [review]

File: tests/unit/ray_tracing_positive.cpp [hunk 1]
```diff
@@ -484,7 +484,7 @@ TEST_F(PositiveRayTracing, AccelerationStructuresOverlappingMemory) {
         for (size_t i = 0; i < blas_count; ++i) {
             auto blas = vkt::as::blueprint::BuildGeometryInfoSimpleOnDeviceBottomLevel(*m_device);
             blas.SetScratchBuffer(scratch_buffer);
-            blas.SetDeviceScratchOffset(i * 8192);
+            blas.SetDeviceScratchOffset(i * (alloc_info.allocationSize / blas_count));
```
Comment (spencer-lunarg) [inline tests/unit/ray_tracing_positive.cpp:487]: Validation Error: [ VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03671 ] Object 0: handle = 0x61670558f9d0, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xf56c9b0000000004, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0x2229b56c | vkCmdBuildAccelerationStructuresKHR(): pInfos[2].scratchData.deviceAddress (0xffff800100040000) has no buffer associated to it such that valid usage passes. At least one buffer associated to this device address must be valid. The following buffers have an address range that does not include scratch address range [0xffff800100040000, 0xffff800100050088):
Object 1: buffer address range is [0xffff800100020000, 0xffff800100050000)

Comment (arno-lunarg) [reply tests/unit/ray_tracing_positive.cpp]: I pushed what should be a fix!

File: tests/unit/ray_tracing_positive.cpp [hunk 2]
```diff
@@ -538,7 +541,7 @@ TEST_F(PositiveRayTracing, AccelerationStructuresReuseScratchMemory) {
         // Create scratch buffer
         VkBufferCreateInfo scratch_buffer_ci = vku::InitStructHelper();
         scratch_buffer_ci.sharingMode = VK_SHARING_MODE_EXCLUSIVE;
-        scratch_buffer_ci.size = 8192;
+        scratch_buffer_ci.size = alloc_info.allocationSize;
```
Comment (spencer-lunarg) [inline tests/unit/ray_tracing_positive.cpp:544]: Validation Error: [ VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03671 ] Object 0: handle = 0x555557c73f00, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xcb3ee80000000007, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0x2229b56c | vkCmdBuildAccelerationStructuresKHR(): pInfos[0].scratchData.deviceAddress (0xffff800100020000) has no buffer associated to it such that valid usage passes. At least one buffer associated to this device address must be valid. The following buffers have an address range that does not include scratch address range [0xffff800100020000, 0xffff800100030088):
Object 1: buffer address range is [0xffff800100020000, 0xffff800100030000)

File: tests/unit/ray_tracing_positive.cpp [hunk 3]
```diff
@@ -572,7 +575,7 @@ TEST_F(PositiveRayTracing, AccelerationStructuresReuseScratchMemory) {
...
```
Comment (spencer-lunarg) [inline tests/unit/ray_tracing_positive.cpp:578]: The next two also have an error, but might be from the first error

Validation Error: [ VUID-vkCmdBuildAccelerationStructuresKHR-pInfos-03671 ] Object 0: handle = 0x555557c77c80, type = VK_OBJECT_TYPE_COMMAND_BUFFER; Object 1: handle = 0xcb3ee80000000007, type = VK_OBJECT_TYPE_BUFFER; Object 2: handle = 0xe7e6d0000000000f, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0x2229b56c | vkCmdBuildAccelerationStructuresKHR(): pInfos[0].scratchData.deviceAddress (0xffff800100020000) has no buffer associated to it such that valid usage passes. At least one buffer associated to this device address must be valid. The following buffers have an address range that does not include scratch address range [0xffff800100020000, 0xffff800100030088):
Object 1: buffer address range is [0xffff800100020000, 0xffff800100030000)

Object 2: buffer address range is [0xffff800100020000, 0xffff800100030000)

Comment (arno-lunarg) [reply tests/unit/ray_tracing_positive.cpp:578]: I bumped memory size, this test should be working now

**Comment by arno-lunarg** [comment]
nice

**Comment by spencer-lunarg** [comment]
I see a crash in `PositiveRayTracing.AccelerationStructuresOverlappingMemory` on NVIDIA, going to re-run and see if consistent

**Comment by spencer-lunarg** [comment]
so seem `AccelerationStructuresOverlappingMemory` was failing prior to this change https://github.com/LunarG/VulkanTests/commit/746a44b0ed638fa520131dc13e49b17a6ce054f5

**Pull request merged**